app/src/main/java/org/p2p/wallet/striga/ui/countrypicker/StrigaCountryPickerFragment.kt

```diff
@@ -0,0 +1,94 @@
+package org.p2p.wallet.striga.ui.countrypicker
+
+import androidx.core.os.bundleOf
+import androidx.core.view.isVisible
+import androidx.fragment.app.setFragmentResult
+import android.os.Bundle
+import android.view.View
+import org.koin.android.ext.android.inject
+import org.koin.core.parameter.parametersOf
+import org.p2p.uikit.model.AnyCellItem
+import org.p2p.wallet.R
+import org.p2p.wallet.auth.repository.Country
+import org.p2p.wallet.auth.widget.AnimatedSearchView
+import org.p2p.wallet.common.adapter.CommonAnyCellAdapter
+import org.p2p.wallet.common.mvp.BaseMvpFragment
+import org.p2p.wallet.databinding.FragmentStrigaCountryPickerBinding
+import org.p2p.wallet.striga.ui.countrypicker.delegates.strigaCountryDelegate
+import org.p2p.wallet.striga.ui.countrypicker.delegates.strigaHeaderDelegate
+import org.p2p.wallet.utils.args
+import org.p2p.wallet.utils.emptyString
+import org.p2p.wallet.utils.popBackStack
+import org.p2p.wallet.utils.viewbinding.getColor
+import org.p2p.wallet.utils.viewbinding.viewBinding
+import org.p2p.wallet.utils.withArgs
+
+typealias IView = StrigaCountryPickerContract.View
+typealias IPresenter = StrigaCountryPickerContract.Presenter
+
+private const val EXTRA_KEY = "EXTRA_KEY"
+private const val EXTRA_RESULT = "EXTRA_RESULT"
+private const val EXTRA_SELECTED_COUNTRY = "EXTRA_SELECTED_COUNTRY"
+
+class StrigaCountryPickerFragment : BaseMvpFragment<IView, IPresenter>(R.layout.fragment_striga_country_picker), IView {
+
+    companion object {
+        fun create(
+            selectedCountry: Country?,
+            requestKey: String,
+            resultKey: String
```

Comment:
i think we have replaceFragmentWithResult from @eduardmaximovich 
try it

Reply:
We have, it's called `replaceFragmentForResult` and it is located in `org.p2p.wallet.utils.NavigationExtensions.kt`.
Btw, preposition `with` is not correct in this case, should be `replaceForResult` (see startActivityForResult)